source/common/http/conn_manager_impl.cc

```diff
@@ -1314,8 +1333,12 @@ void ConnectionManagerImpl::ActiveStream::setBufferLimit(uint32_t new_limit) {
 }
 
 bool ConnectionManagerImpl::ActiveStream::createFilterChain() {
+  if (state_.created_filter_chain_) {
+    return false;
+  }
   bool upgrade_rejected = false;
-  auto upgrade = request_headers_->Upgrade();
+  auto upgrade = request_headers_ ? request_headers_->Upgrade() : nullptr;
```

Comment:
Is it possible to not have headers here? Is the case where someone somehow calls sendLocalReply before decodeHeaders() is called? Is that possible?

Reply:
I believe in the HTTP/1.1 path we call newStream onMessageBegin, and arm the request timer there, so if we time out when reading headers this might happen.  I suspect the idle timer should cover from stream creation until destruction as well though it looks like that one only arms when the read filter callbacks are created which seems like a bug to me.

Reply:
Hah great I was going to ask you to add a test for this case, but you are always ahead of me. :)